### ISSUE TITLE
Save/Load functionality

### ISSUE THREAD

**Issue by keisetsu** [RabbitMQSimulator/RabbitMQSimulator#13]
It would be very useful to be able to save to and load from a file, say .json. I guess this could be done using rabbit, but it seems like it should be possible to do this without rabbit installed at all.

**Comment by videlalvaro** [comment]
there's export to JSON, and the "player" feature kinda does what you want. I will try to add this

**Comment by andrewmichaelsmith** [comment]
Is there a way to take the "To Player" output and import it that would be perfect. I built up something on tryrabbitmq.com and wanted to move it to where I was running rabbitmq locally but this just doesn't seem to be possible.

**Comment by videlalvaro** [comment]
So the undocumented "To Player" thing, will create a JSON export that's readable by the simulator Player, not by RabbitMQ. If you run the simulator locally and connect it to the management plugin (sadly I think you will need to modify the user password in the simulator code), then you can export/import from/to rabbitmq

**Comment by andrewmichaelsmith** [comment]
What I want is a mechanism to load that JSON in to the simulator (not rabbitmq). 

I wanted to use this to document our internal use of rabbitmq (consisting of many consumers, producers, exchanges and queues). To do this I would like to have the JSON describing the simulation in version control, then when we make changes we can re-load this JSON, make our modifications and commit it. Rather than having to manually re-create it.

I was able to get the "To RabbitMQ" and "From RabbitMQ" set up working (which is really nice!). And I see how I could achieve persistence by doing this. But there's no way for me to version control (or easily share) this, I would being using RabbitMQ as my persistence mechanism.

**Comment by videlalvaro** [comment]
That's what the "Player" mode is for. Take a look at https://github.com/RabbitMQSimulator/RabbitMQSimulator/blob/master/web/js/tutorial_1.js and the various player*.html views on how to use this. I grant you that the current status is not very modular

**Comment by andrewmichaelsmith** [comment]
Player mode can read the JSON but can I make subsequent changes in player mode?

**Comment by videlalvaro** [comment]
No. You can't do that. That's a good point. So basically we need a way to persist changes and be able to load those, probably from the simulator UI

**Comment by andrewmichaelsmith** [comment]
Yes ideally, personally I wouldn't minding using to "To Json" as the save function, then it would just be a "From Json" that is missing.
